.github/workflows/ci.yml

```diff
@@ -137,6 +137,12 @@ jobs:
           command: test
           args: --features "block_on executor"
 
+      - name: Run tests with signals
+        uses: actions-rs/cargo@v1
+        with:
+          command: test
+          args: --features "block_on executor signals"
+
```

Comment:
Why do this second test step in rather than just enabling the `signals` feature on the previous entry?

Reply:
In order to test the absence of `nix`, which was an issue while I was testing it.

It can be an issue sometimes where  some permutations of features have compile errors without the maintainer knowing. It might be a good idea to use [`cargo-hack`](https://github.com/taiki-e/cargo-hack) in the CI for this crate.